### ISSUE TITLE
Chaining into rpt2 from a non-.ts file?

### ISSUE THREAD

**Issue by coolreader18** [ezolenko/rollup-plugin-typescript2#111]
#### What happens and why it is wrong
I'm compiling [nearley](https://github.com/kach/nearley) to typescript and trying to pass it through to rpt2, but it's not working. The plugin just returns the compiled code, and I tried adding `["*.ne", "**/*.ne"]` to the rpt2 options, but just doing that wasn't working, giving an error `Error: Could not find file: '/home/coolreader18/Projects/betterfunction/src/lib/grammar/index.ne'.`. After doing some debugging in typescript, I figured out that it was because it wasn't including the .ne file in the compilation, and it was erroring in the `service.getEmitOutput` call in [index.ts L195](https://github.com/ezolenko/rollup-plugin-typescript2/blob/3840a25e6e94a9271a3cb20e3938f8782412f2c2/src/index.ts#L195). Is there maybe an alternate call to use to get the output from a file? Before trying to refactor it into chaining, I just called `typescript().transform.call(this, code, id + ".ts")` from the nearley plugin, and that worked. Maybe error handling for that call, and on an error it just uses the source code provided as a parameter?

#### Versions
- typescript: 3.1.0-dev.20180825
- rollup: 0.65.0
- rollup-plugin-typescript2: 0.16.1

#### rollup.config.js
https://gist.github.com/coolreader18/c15cb2e0e8483d6c69769a0b8ade20a3
Previously I was doing:
```js
const code = generate(compilation, path.basename(id));
const ret = ts.transform.call(this, code, `${id}.ts`);
return ret;
```
in the transform function, but I think chaining is a lot cleaner and less hacky.

#### tsconfig.json
```json
{
  "compilerOptions": {
    "module": "es2015",
    "strictNullChecks": true,
    "target": "es2017",
    "lib": ["es2017"],
    "moduleResolution": "node"
  }
}
```

#### plugin output with verbosity 3
Full error message:
```
[!] (rpt2 plugin) Error: Could not find file: '/home/coolreader18/Projects/betterfunction/src/lib/grammar/index.ne'.
src/lib/grammar/index.ne
Error: Could not find file: '/home/coolreader18/Projects/betterfunction/src/lib/grammar/index.ne'.
    at getValidSourceFile (/home/coolreader18/Projects/betterfunction/node_modules/typescript/lib/typescript.js:112719:23)
    at Object.getEmitOutput (/home/coolreader18/Projects/betterfunction/node_modules/typescript/lib/typescript.js:113077:30)
    at cache.getCompiled (/home/coolreader18/Projects/betterfunction/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:20101:40)
    at TsCache.getCompiled (/home/coolreader18/Projects/betterfunction/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:19765:33)
    at Object.transform (/home/coolreader18/Projects/betterfunction/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:20099:36)
    at /home/coolreader18/Projects/betterfunction/node_modules/rollup/dist/rollup.js:20831:25
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:182:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:697:11)
    at startup (internal/bootstrap/node.js:201:19)
```

**Comment by ezolenko** [comment]
@coolreader18 did you try adding `"*.ne", "**/*.ne"` to include part in tsconfig itself?

**Comment by ezolenko** [comment]
Could you make a small repo with reproduction (or post a link to your full repo if it is opensource)

**Comment by coolreader18** [comment]
Adding it to tsconfig.json didn't help, it's giving the same error. The repo is [coolreader18/betterfunction](https://github.com/coolreader18/betterfunction).

**Comment by chinesedfan** [comment]
I have analyzed this problem recently. Refer to https://github.com/tsne/rollup-plugin-tsc/issues/4#issuecomment-429906319.

**Comment by ezolenko** [comment]
@coolreader18 I looked at what happens again (using your repo at last aug 31st commit) and it seems like rollup never tries to pass *.ne files to rpt2 at all, imports got converted to requires and left that way. Either I reconstructed the problem incorrectly, or `nearleyPlugin` needs to implement `resolveId` and maybe `load`.

@chinesedfan could you make an example repo that reproduces this?

**Comment by chinesedfan** [comment]
@ezolenko In fact, it is [the extra filter](https://github.com/tsne/rollup-plugin-tsc/commit/d4a89a8a4b7e2f148e98441c7f711f2672d4e761#diff-480db08609ef6a51bf20f049cc9bda48R77), not `allowNonTsExtensions`, that fixes the problem in my repository [chinesedfan/rollup-pipe-ts-plugin-issue](https://github.com/chinesedfan/rollup-pipe-ts-plugin-issue).

**Comment by ezolenko** [comment]
@chinesedfan does current master work for you (if you set correct include in both tsconfig and plugin include option)? Not sure I want to always pass everything that is in tsconfig, it is tempting, but I'm not sure how to treat plugin's own include/exclude option then.

**Comment by chinesedfan** [comment]
@ezolenko Thanks. It works.

I noticed that rpt2 is different with rollup-plugin-tsc. It uses filter created by plugin include/exclude, instead language service filter. And `allowNonTsExtensions` is also important.

**Comment by ezolenko** [comment]
Fixed in 0.18.0
